File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/TaxCategoryColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TaxCategoryRow } from "./TaxCategoryRow";

export interface TaxCategoryColumns {
    TaxCategoryId: Column<TaxCategoryRow>;
    SystemCode: Column<TaxCategoryRow>;
    SystemName: Column<TaxCategoryRow>;
    UserCode: Column<TaxCategoryRow>;
    UserName: Column<TaxCategoryRow>;
}

export class TaxCategoryColumns extends ColumnsBase<TaxCategoryRow> {
    static readonly columnsKey = 'CrmDB.TaxCategory';
    static readonly Fields = fieldsProxy<TaxCategoryColumns>();
}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/TaxCategory/TaxCategoryColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace CultureLinkCRM.CrmDB.Columns;

[ColumnsScript("CrmDB.TaxCategory")]
[BasedOnRow(typeof(TaxCategoryRow), CheckNames = true)]
public class TaxCategoryColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int TaxCategoryId { get; set; }
    [EditLink]
    public string SystemCode { get; set; }
    public string SystemName { get; set; }
    public string UserCode { get; set; }
    public string UserName { get; set; }
}